File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cpp
#include <string>
#include <vector>
#include <set>
using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
    vector<int> answer;
    set<int> res;
    for(auto a : arr)
    {
        if(a%divisor == 0) res.insert(a);
    }
    // 문제가 원하는 반환형이 vector<int>라서 {-1}
    if(res.empty()) return {-1};
    
    for(auto r :res)
    {
        answer.push_back(r);
    }
    
    return answer;
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.cs
using System.Linq;
using System;
public class Solution {
    public int[] solution(int[] arr, int divisor) {
        Array.Sort(arr);
        int[] answer = arr.Where(x=>x%divisor == 0).ToArray();
        if(answer.Length == 0) answer = new int[] { -1 };
        return answer;
    }
}